### PR TITLE
KillTask should return empty inputSource resources

### PR DESCRIPTION
**Pull request by zachjsh**
### Description

This pr fixes a few bugs found with the inputSource security feature.

1. `KillUnusedSegmentsTask` previously had no definition for the `getInputSourceResources`, which caused an unsupportedOperationException to be thrown when this task type was submitted with the inputSource security feature enabled. This task type should not require any input source specific resources, so returning an empty set for this task type now.

2. Fixed a bug where when the input source type security feature is enabled, all of the input source type specific resources used where authenticated against:

`{"resource": {"name": "EXTERNAL", "type": "{INPUT_SOURCE_TYPE}"}, "action": "READ"}`

When they should be instead authenticated against:

`{"resource": {"name": "{INPUT_SOURCE_TYPE}", "type": "EXTERNAL"}, "action": "READ"}`

3. fixed bug where supervisor tasks were not authenticated against the specific input source types used, if input source security feature was enabled.

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java
Comment (abhishekrb19) [inline indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java:205]: Nit: there's an extra whitespace:
```suggestion
    return Collections.singleton(new ResourceAction(new Resource("hadoop", ResourceType.EXTERNAL), Action.READ));
```

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java
Comment (abhishekrb19) [inline indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java:205]: I think Hadoop's task type is called `index_hadoop` (and not `hadoop` - see https://druid.apache.org/docs/latest/ingestion/hadoop.html#task-syntax and the method `getType()` above).

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java
Comment (abhishekrb19) [inline indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java:257]: To avoid any confusion, I think it'd be better to rename all references of the tests that now invoke `getInputSourceResources` (previously `getInputSourceTypes`):
 - `testCorrectInputSourceTypes` -> `testCorrectInputSourceResources`?
- `hasCorrectInputSourceTypes ` -> `hasCorrectInputSourceResources`
...and so on

File: indexing-service/src/test/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResourceTest.java
Comment (abhishekrb19) [inline indexing-service/src/test/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResourceTest.java:228]: 👍

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/HadoopIndexTaskTest.java
Comment (abhishekrb19) [inline indexing-service/src/test/java/org/apache/druid/indexing/common/task/HadoopIndexTaskTest.java:75]: Perhaps this could be a static variable in `HadoopIndexTask` that can be reused here? It'd also clarify the difference between task type and input source type:
```
  public static final String INPUT_SOURCE_TYPE = "hadoop";
```

Comment (zachjsh) [reply indexing-service/src/test/java/org/apache/druid/indexing/common/task/HadoopIndexTaskTest.java]: good suggestion, fixed

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java
Comment (zachjsh) [reply indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java:257]: fixed

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java
Comment (zachjsh) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java]: fixed

**Review by abhishekrb19 — APPROVED** [review]
Thanks for the fix and addressing the review feedback! LGTM after CI
Should we backport this fix to 26.0 once it's merged to master? (cc: @clintropolis)

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java
Comment (abhishekrb19) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java:153]: Static analysis is failing with the `addAll` call. Think it can be done inline with the stream() chain:
```java
final Set<ResourceAction> resourceActions = spec.getDataSources().stream()
                                                    .map(dataSource -> new ResourceAction(new Resource(
                                                        dataSource,
                                                        ResourceType.DATASOURCE
                                                    ), Action.WRITE)).collect(Collectors.toSet());
```

**Pull request merged**
